iron-dropdown.html

```diff
@@ -340,6 +340,49 @@
           } else {
             Polymer.IronOverlayBehaviorImpl._applyFocus.apply(this, arguments);
           }
+        },
+
+        /**
+         * An element related to the dropdown (the triggering element, or an
+         * element within the dropdown) has lost focus.
+         */
+        _activeElementBlurred: function(event) {
+          if (event.target._blurListener) {
+            // Stop listening to blur events on the target.
+            event.target.removeEventListener('blur', event.target._blurListener);
+            event.target._blurListener = null;
+          }
+          this._dropdownFocusChanged(false);
+        },
+
+        /**
+         * Return true if the focus is anywhere within the dropdown.
+         */
+        _focusWithinDropdown: function() {
+          if (this.shadowRoot) {
+            return this.shadowRoot.activeElement !== null;
+          } else {
+            var activeElement = this._manager.deepActiveElement;
+            return Polymer.dom(this).deepContains(activeElement);
+          }
+        },
+
+        /**
+         * Overridden from `IronOverlayBehavior`.
+         */
+        _onCaptureFocus: function(event) {
+          if (!this._focusWithinDropdown()) {
```

Comment:
`_focusWithinDropdown` might be suboptimal as it first searches for the deepest focused element & then looks if it is deeply contained in the dropdown.
Why not using `event.path` and checking if `this` is in that array? 
```javascript
var path = Polymer.dom(event).path;
if (path.indexOf(this) === -1) {
```